docs/MPURegionsUsageGuide.md

```diff
@@ -0,0 +1,299 @@
+		```
+
+#### Configuring MPU Regions for Common Binary
+- A Common Binary contains shared code b/w all apps. This feature is enabled with CONFIG_SUPPORT_COMMON_BINARY.
+- MPU Region settings are configured at time of loading common binary (inside load_binary function).
+- Below 2 cases arise for MPU region configuration for common binary:
+	1. **CONFIG_OPTIMIZE_APP_RELOAD_TIME=y**
+		- In this case, 3 MPU regions are used for common binary.
+		- Below is MPU region info for this case:
+		```
+		----------------------------------------------------------------------------
+		MPU Region			Memory Segments		Permissions
+		----------------------------------------------------------------------------
+		MPU_REG_NUM_COM_LIB_TXT		TEXT			ROX
+		MPU_REG_NUM_COM_LIB_RO		RO			RONX
+		MPU_REG_NUM_COM_LIB_DATA	Complete RAM partition	RWNX
+		----------------------------------------------------------------------------
+		```
+	2. **CONFIG_OPTIMIZE_APP_RELOAD_TIME=n**
+		- In this case, only 1 MPU region is used for entire RAM address range for common binary.
+		- Below is MPU region info in this case:
+		```
+		----------------------------------------------------------------------------
+		MPU Region		Memory Segments		Permissions
+		----------------------------------------------------------------------------
+		MPU_REG_NUM_COM_LIB	Complete RAM partition	RWNX
+		----------------------------------------------------------------------------
+		```
+
+#### Configuring MPU region for Stack Overflow Protection
+- This feature is enabled with CONFIG_MPU_STACK_OVERFLOW_PROTECTION.
+- Just 1 MPU region (MPU_REG_NUM_STK) is shared amongst all tasks that are created.
+- Stack Allocation for each task includes CONFIG_MPU_STACK_GUARD_SIZE bytes of Guard Memory reserved.
+- At time of context switch, MPU_REG_NUM_STK region assigns RONX permission to this Guard Memory (inside up_create_task function).
+- MPU region settings are saved in Task Control Block Structure (struct tcb_s \*) inside mpu_stack_regs[] member.
+- Below is MPU region info in this case:
+```
+--------------------------------------------------------------------------
+MPU Region		Memory Segment		 	Permissions
+--------------------------------------------------------------------------
+MPU_REG_NUM_STK		Task Stack Guard Memory		RONX
+--------------------------------------------------------------------------
+```
+
+## Sample memory layout of MPU regions
+
+- Below diagram is an example of memory layout for MPU regions.
+- For this example, below Boolean configs were enabled:
+	1. CONFIG_APP_BINARY_SEPARATION
+	2. CONFIG_OPTIMIZE_APP_RELOAD_TIME
+	3. CONFIG_SUPPORT_COMMON_BINARY
+	4. CONFIG_MPU_STACK_OVERFLOW_PROTECTION
+
+```
+		RAM
+
+	+===============+
+	|	 	|
+	|		|
+Kernel	|  Kheap/KStack |
+Heap	|		|
+	|===============|
+	|		|
+	|		|  MPU_APP_TEXT(ROX) MPU_APP_RO(RONX) MPU_APP_DATA(RWNX)    MPU_STACK(RO)      RWNX            RO            RWNX
```

Comment:
For stack, you write RONX at previous, but here is RO. Could you use the same attr?

Reply:
done